### PR TITLE
[CK_TILE][FMHA] Add new tile size for async

### PR DESCRIPTION
**Pull request by LJ-underdog**
## Proposed changes

This PR adds a new tile size configuration for async pipeline operations in the FMHA (Fused Multi-Head Attention) forward pass implementation. The changes introduce support for a 64x128 tile configuration for specific head dimension scenarios and adjust the sequence tuning logic to accommodate this new tile size.
Changes:

Modified sequence tuning logic to include tile size 64 as a special case alongside the maximum tile size
Added filtering logic to exclude 64-size tiles for non-async pipelines with 128x128 head dimensions
Introduced a new 64x128x32 tile size configuration with compute unit constraint for 128x128 head dimensions
## Checklist

Please put an `x` into the boxes that apply. You can also fill these out after creating the PR. If you're not sure, please don't hesitate to ask.

- [x] I have added tests relevant to the introduced functionality, and the unit tests are passing locally
- [x] I have added the test to REGRESSION_TESTS list defined at the top of CMakeLists.txt in tests/CMakeLists.txt, **IF** the test takes more than 30 seconds to run.
- [x] I have added inline documentation which enables the maintainers with understanding the motivation
- [x] I have removed the stale documentation which is no longer relevant after this pull request
- [x] (If this change is user-facing) I have added release notes which provide the end users with a brief summary of the improvement from this pull request
- [x] I have run `clang-format` on all changed files
- [x] Any dependent changes have been merged

## Discussion

If this is a relatively large or complex change, feel free to start a discussion by explaining why you chose the solution you did and what alternatives you considered

### CODE REVIEW

**Comment by LJ-underdog** [comment]
Performance` improve 30% on gfx950 when block_num <= cu_num

`./bin/tile_example_fmha_fwd -h=8  -d=128 -s=512 -kname=1 -v=1
[fp16|batch|bhsd] b:2, h:8/8, s:512/512, d:128/128, scale_s:0.0883883, bias:n, p_drop:0, lse:0, qscale:n, mask:n, v:r, fmha_fwd_d128_fp16_batch_b128x64x32x128x16x128_r4x1x1_r4x1x1_w32x32x16_w32x32x16_qr_async_trload_vr_npad_nlogits_nbias_nmask_nlse_ndropout_nskip_nqscale_trload_nsink, 0.020 ms, 109.11 TFlops, 426.21 GB/s, valid:y
 
./bin/tile_example_fmha_fwd -h=8  -d=128 -s=512 -kname=1 -v=1
[fp16|batch|bhsd] b:2, h:8/8, s:512/512, d:128/128, scale_s:0.0883883, bias:n, p_drop:0, lse:0, qscale:n, mask:n, v:r, fmha_fwd_d128_fp16_batch_b64x128x32x128x32x128_r4x1x1_r4x1x1_w16x16x32_w16x16x16_qr_async_vr_psddv_nlogits_nbias_nmask_nlse_ndropout_nskip_nqscale_ntrload_nsink, 0.015 ms, 143.74 TFlops, 561.49 GB/s, valid:y`

**Review by illsilin — APPROVED** [review]
